File: 08_Linked_Lists/Remove_duplicates_from_sorted_list/remove_duplicates_from_sorted_list.cpp
// LeetCode Problem 83 - https://leetcode.com/problems/remove-duplicates-from-sorted-list/     
// TC - O(n), SC - O(1)

ListNode* deleteDuplicates(ListNode* head) {
        if(!head) return head;
        ListNode* curr = head;
        while(curr->next !=NULL){
            if(curr->val == curr->next->val){
                curr->next = curr->next->next;
            }
            else{
                curr = curr->next;
            }
        }
        return head;
    }

File: 08_Linked_Lists/Remove_duplicates_from_sorted_list/remove_duplicates_from_sorted_list.java
// LeetCode Problem 83 - https://leetcode.com/problems/remove-duplicates-from-sorted-list/     
// TC - O(n), SC - O(1)

class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;

        ListNode curr = head;

        while (curr.next != null) {
            if (curr.val == curr.next.val) {
                // Skip duplicate node
                curr.next = curr.next.next;
            } else {
                curr = curr.next;
            }
        }
        return head;
    }
}